### PR TITLE
Spammy log when one bookie of ensemble is down #2285

### LINKED ISSUE #2285: Spammy log when one bookie of ensemble is down
When one bookie of the ensemble is down, the others bookies client  spam in the log ***org.apache.bookkeeper.proto.PerChannelBookieClient*** this exception: 

```
20-03-10-10-42-01       Could not connect to bookie: [id: 0x823ef32e, L:/10.162.1.12:42330]/es-dev1.sviluppo.dna:3181, current state CONNECTING :
20-03-10-10-42-01       io.netty.channel.AbstractChannel$AnnotatedConnectException: syscall:getsockopt(..) failed: Connection refused: es-dev1.sviluppo.dna/10.162.1.11:3181
io.netty.channel.AbstractChannel$AnnotatedConnectException: syscall:getsockopt(..) failed: Connection refused: es-dev1.sviluppo.dna/10.162.1.11:3181
        at io.netty.channel.unix.Socket.finishConnect(..)(Unknown Source)
Caused by: io.netty.channel.unix.Errors$NativeConnectException: syscall:getsockopt(..) failed: Connection refused
        ... 1 more 

```
In our case, in 8m it produced 1,5gb log.
For riproduced it, i simple shutdown a bookie and checked **org.apache.bookkeeper.proto.PerChannelBookieClient**

### PR DESCRIPTION
**Pull request by nicoloboschi**
Descriptions of the changes in this PR:

### Motivation

Resolves #2285

### Changes

We write "bookie server unavailable log" every 30 seconds per bookie client

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java [hunk 1]
```diff
@@ -2488,6 +2490,14 @@ public void operationComplete(ChannelFuture future) throws Exception {
 
             makeWritable();
         }
+
+        private void logBookieUnavailable(Consumer<Void> log) {
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:2494]: why `Consumer<Void> log` and not `Runnable` ?

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java [hunk 2]
```diff
@@ -2488,6 +2490,14 @@ public void operationComplete(ChannelFuture future) throws Exception {
...
+            final long now = System.currentTimeMillis();
+            if ((now - lastBookieUnavailableLog) > 30_000) {
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:2496]: we should make this 30_000 configurable or calculate it from other timeout configuration parameters

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java [hunk 3]
```diff
@@ -331,6 +332,7 @@
     private final ExtensionRegistry extRegistry;
     private final SecurityHandlerFactory shFactory;
     private volatile boolean isWritable = true;
+    private long lastBookieUnavailableLog = 0;
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:335]: what about renaming it to `lastBookieUnavailableLogTimestamp` ?

**Review by sijie — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM
merging as soon as CI is green

**Comment by eolivelli** [comment]
committed to master branch, thank you @nicoloboschi 

If you want  please create a separate patch for branch-4.11

**Pull request merged**